swap_meet/vendor.py

```diff
@@ -1,2 +1,72 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+
+    def get_by_category(self, category):
+        items_by_category = [item for item in self.inventory if item.category == category]
+        return items_by_category
+
+    def swap_items(self, other, yours, theirs):
+        if yours in set(self.inventory) and theirs in set(other.inventory):
+            other.add(yours)
+            self.remove(yours)
+            self.add(theirs)
+            other.remove(theirs)
+            return True
```

Comment:
The specification for this function is to either return `True` (successful swap) or `False` (no swap). However, as written, this will return `True` on success, but `None` on no swap, since there's no `return` for the error case. Even if we _did_ want to return `None`, since we're explicitly returning in one code path, we should do so in all of them, but here, we should really be returning `False`.

Additionally, if we invert the error check, we can use it as a guard clause so that we don't need to indent the main flow of the function.

```python
    def swap_items(self, other, yours, theirs):
        if yours not in self.inventory or theirs not in other.inventory:
            return False

        other.add(yours)
        self.remove(yours)
        self.add(theirs)
        other.remove(theirs)
        return True
```